src/partitioner/histogram.py

```diff
@@ -60,3 +60,74 @@ def generate_partial_histogram(
     mapped_pixel, count_at_pixel = np.unique(mapped_pixels, return_counts=True)
     histo[mapped_pixel] += count_at_pixel.astype(np.ulonglong)
     return histo
+
+
+def generate_alignment(histogram, highest_order=10, threshold=1_000_000):
+    """Generate alignment from high order pixels to those of equal or lower order
+
+    Note:
+        We may initially find healpix pixels at order 10, but after aggregating up to the pixel
+        threshold, some final pixels are order 4 or 7. This method provides a map from pixels
+        at order 10 to their destination pixel. This may be used as an input into later partitioning
+        map reduce steps.
+    Args:
+        histogram (:obj:`np.array`): one-dimensional numpy array of long integers where the
+            value at each index corresponds to the number of objects found at the healpix pixel.
+        highest_order (int):  the highest healpix order (e.g. 0-10)
+        threshold (int): the maximum number of objects allowed in a single pixel
+    Returns:
+        one-dimensional numpy array of integer 3-tuples, where the value at each index corresponds
+        to the destination pixel at order less than or equal to the `highest_order`.
+        The tuple contains three integers:
+            [0]: order of the destination pixel
+            [1]: pixel number *at the above order*
+            [2]: the number of objects in the pixel
+    """
+
+    if len(histogram) != hp.order2npix(highest_order):
+        raise ValueError("histogram is not the right size")
+
+    nested_sums = []
+    for i in range(0, highest_order):
+        nested_sums.append(empty_histogram(i))
+    nested_sums.append(histogram)
+
+    # work backward - from highest order, fill in the sums of lower order pixels
+    for read_order in range(highest_order, 0, -1):
+        parent_order = read_order - 1
+        for index in range(0, len(nested_sums[read_order])):
+            parent_pixel = index >> 2
+            # print(f"{read_order} : {index} : {parent_order} : {parent_pixel}")
```

Comment:
do you want to keep this debug statement (?) around?

Reply:
No I do not!